### ISSUE TITLE
IncompatibleClassChangeError on android 4.0.4

### ISSUE THREAD

**Issue by p5n** [nextcloud/android#1090]
### Actual behaviour
Nextcloud app crashes after some time without any user action. I've googled and found this solution: 

https://stackoverflow.com/questions/10797551/incompatibleclasschangeerror-interface-not-implemented

They say that you need to add android-support-v4.jar to libs. I did not test it yet because of it need application build from sources as I understand

Previous version (1.4.1 from F-Droid) worked for me.

### Steps to reproduce
1. Run nextcloud app
2. Just wait for crash

### Environment data
Android version: 4.0.4

Device model: 
Star V1277

Nextcloud app version:
Both 1.4.2 from F-Droid and https://download.nextcloud.com/android/nightly/nextcloud-dev-20170614.apk crashes

Nextcloud server version:
12.0.0

### Logs
```
E/AndroidRuntime( 3223): FATAL EXCEPTION: IntentService[JobRescheduleService]
E/AndroidRuntime( 3223): java.lang.IncompatibleClassChangeError: interface not implemented
E/AndroidRuntime( 3223):        at com.evernote.android.job.JobStorage.closeSilently(JobStorage.java:453)
E/AndroidRuntime( 3223):        at com.evernote.android.job.JobStorage.getAllJobRequests(JobStorage.java:195)
E/AndroidRuntime( 3223):        at com.evernote.android.job.JobRescheduleService.onHandleIntent(JobRescheduleService.java:44)
E/AndroidRuntime( 3223):        at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
E/AndroidRuntime( 3223):        at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 3223):        at android.os.Looper.loop(Looper.java:154)
E/AndroidRuntime( 3223):        at android.os.HandlerThread.run(HandlerThread.java:65)
D/dalvikvm(  243): threadid=85: interp stack at 0x4d263000
W/ActivityManager(  243):   Force finishing activity com.nextcloud.client/com.owncloud.android.ui.activity.FileDisplayActivity
```

See also:
https://gitlab.com/fdroid/fdroiddata/issues/754

**Comment by mario** [comment]
Thanks for your report. This will be fixed in the follow-up stable release.

**Comment by tobiasKaminsky** [comment]
https://github.com/evernote/android-job/issues/182

According to this it can be changed with using 1.1.10.
I did a small test on my very old phone and it works.

@mario do you see any with enhancing the version?

**Comment by mario** [comment]
@tobiasKaminsky I'm going to update it in the media branch, so don't bother with this for now.

**Comment by tobiasKaminsky** [comment]
I thought to upgrade this in dev version manually to have a an extended test of the library?

**Comment by tobiasKaminsky** [comment]
I bumped the version on dev.
For master we still need to change this.
@p5n if you want to can test this in the next days with a new dev version.

**Comment by p5n** [comment]
https://download.nextcloud.com/android/dev/nextcloud-dev-20170622.apk still fails

**Comment by tobiasKaminsky** [comment]
Well... I did this 2 hours ago and the build is from this night ;-)
You can try it next day.

**Comment by p5n** [comment]
https://download.nextcloud.com/android/dev/nextcloud-dev-20170623.apk works slightly better. I was able to set up account, but it failed when I tap "settings"
```
E/AndroidRuntime( 8364): FATAL EXCEPTION: main
E/AndroidRuntime( 8364): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nextcloud.android.beta/com.owncloud.android.ui.activity.Preferences}: java.lang.ClassCastException: com.owncloud.android.ui.CheckBoxPreferenceWithLongTitle cannot be cast to android.preference.SwitchPreference
E/AndroidRuntime( 8364): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2077)
E/AndroidRuntime( 8364): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2104)
E/AndroidRuntime( 8364): 	at android.app.ActivityThread.access$600(ActivityThread.java:134)
E/AndroidRuntime( 8364): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1247)
E/AndroidRuntime( 8364): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 8364): 	at android.os.Looper.loop(Looper.java:154)
E/AndroidRuntime( 8364): 	at android.app.ActivityThread.main(ActivityThread.java:4624)
E/AndroidRuntime( 8364): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime( 8364): 	at java.lang.reflect.Method.invoke(Method.java:511)
E/AndroidRuntime( 8364): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:809)
E/AndroidRuntime( 8364): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:576)
E/AndroidRuntime( 8364): 	at dalvik.system.NativeStart.main(Native Method)
E/AndroidRuntime( 8364): Caused by: java.lang.ClassCastException: com.owncloud.android.ui.CheckBoxPreferenceWithLongTitle cannot be cast to android.preference.SwitchPreference
E/AndroidRuntime( 8364): 	at com.owncloud.android.ui.activity.Preferences.onCreate(Preferences.java:543)
E/AndroidRuntime( 8364): 	at android.app.Activity.performCreate(Activity.java:4479)
E/AndroidRuntime( 8364): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1050)
E/AndroidRuntime( 8364): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2041)
E/AndroidRuntime( 8364): 	... 11 more
```

**Comment by tobiasKaminsky** [comment]
This is solved in #1133

**Comment by p5n** [comment]
nextcloud-dev-20170627.apk looks working, I think this bug can be closed as well as #1133.

**Comment by tobiasKaminsky** [comment]
Thanks for confirmation.
We close this once it is merged into master.
